external-crates/move/crates/move-compiler/src/typing/ast.rs

```diff
@@ -210,6 +224,7 @@ pub enum UnannotatedExp_ {
     },
     NamedBlock(BlockLabel, Sequence),
     Block(Sequence),
+    ExpandedMacro(MacroCallInfo, /* inlined macro body */ Sequence),
```

Comment:
This should hold an `Exp`, not a `Sequence`, in case macros some day don't generate sequences.

In addition, what if we simply called this `IDEAnnotation` and the first field is a `Box<IDEInfo>` which could then be an extensible enum?

Reply:
Hmm, I thought I provided additional info on this but I guess I didn't so here it is. A wrinkle here is that the inlined macro sequence pre-pends by-value arguments to the actual function body. The IDE needs to be able to access these to generated symbols for them correctly. I originally went for `Exp` here but then in the IDE I need to assert that this `Exp` is a block anyway and extract a sequence from it to process by-value args. Leaving the sequence here seemed like a cleaner solution in the end.

Reply:
What if the macro doesn't return a sequence because it takes no arguments?

Macros are logically `Args -> Exp` and it just so happens that they currently return sequences. You could also just propagate the match subject into onto the macro call info, since this is only running in IDE mode.

Reply:
> What if the macro doesn't return a sequence because it takes no arguments?

This will result in an empty sequence,`MacroCallInfo`'s `by_value_args_num` value would be 0, and IDE would have nothing to iterate over.

> Macros are logically `Args -> Exp` and it just so happens that they currently return sequences. You could also just propagate the match subject into onto the macro call info, since this is only running in IDE mode.

Do you mean copying `Bind`'s that represent by-value args to `MacroCallInfo` so that we don't have to look for them in the IDE?

Reply:
Replaced `Sequence` with `Exp` and moved by-value args to `MacroCallInfo`, but left the node name instead of introducing `IDEAnnotation`. Since `Exp` in `ExpandedMacro` really represents an inlined macro block (and is generated in IDE mode instead of a 'Block`), I think we should have a more descriptive name for this node rather than a super-generic one such as IDEAnnotation.